src/main/java/org/commcare/formplayer/database/models/EntitiesSelectionStorage.java

```diff
@@ -0,0 +1,85 @@
+package org.commcare.formplayer.database.models;
+
+import static org.commcare.formplayer.sandbox.SqlSandboxUtils.execSql;
+
+import com.google.common.base.Joiner;
+
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.formplayer.sandbox.SqlHelper;
+import org.commcare.modern.database.DatabaseHelper;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+
+/**
+ * A simple sql backed cache that stores the selectons on a multi select Entity list into user DB
+ */
+public class EntitiesSelectionStorage implements EntitiesSelectionCache {
```

Comment:
can you articulate what made you decide to use sqlite for this vs PostgreSQL?

Reply:
I was quite confused to be honest on where should this go and decided to go with Sqlite to have more consistency between mobile and Formplayer architecture wrt this table. That being said, I am curious to hear if there are reasons because of which we should be considering `Postgres` as the primary datastore for FP instead of Sqlite ?

Reply:
I think there are a few things to consider:
1. This further entrenches us in the 'db per machine' world
2. Having it in PG would make it possible to clear old data on a schedule (similar to what we do already with form sessions)

Reply:
Interesting. That's good to know about and makes sense in general. Though in context of this table I still think having ability to use Sqlite on both mobile and FP give us better code sharing opportunities between the two platforms. And since we have a bunch of other Sqlite tables It doesn't seem like adding a simple key value cache for temporal selections should introduce any additional bottlenecks if we are to make a transition at some point. 

Also we can potentially use [same cron task ](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/services/ScheduledTasks.java#L33)to delete the entries from Sqlite table as well. Even if that's not possible, I don't imagine having a different schedule for purging this table should cause an issue.

Let me know if you feel strongly about PG though, it should be an easy change to make.

Reply:
> Also we can potentially use [same cron task ](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/services/ScheduledTasks.java#L33)to delete the entries from Sqlite table as well.

The challenge with this is that it's not a single table so we'd need to query every sqlite DB on the machine individually which is much trickier.

> Though in context of this table I still think having ability to use Sqlite on both mobile and FP give us better code sharing opportunities between the two platforms.

I don't think this is a big issue, and already you've put this code in formplayer repo so it's already not shared.

In general I think the main advantage of the sqlite DBs is that we have very fast IO which is essential for doing the case DB querie. For use cases like this we don't need that since we're just loading a blob of data (and we can cache it in redis if we want).

I'd be curious if @ctsims has any thoughts.

Reply:
Right now I think the split in responsibility is that PG is managing volatile session data (menu sessions, incomplete forms, form session data, etc), and that the sqlite dbs are storing reference data. I think that makes Postgres the most congruous place to store it, since it'll make devops tasks and management the most consistent. If for instance, a sqlite bearing machine is removed from the cluster: Currently a user's form session is retained, and a they continue an existing form session once their newly provisioned sqlite bearing machine syncs their reference data. Putting the data in Postgres would enable the same with this type of data, in addition to the regular cleaning that Simon mentioned, taking advantage of an existing mechanism (the incomplete form clearing db task on startup) without needing a cron-like task.

Reply:
makes sense. Thanks for the inputs @ctsims and @snopoke!

Reply:
implemented changes in https://github.com/dimagi/formplayer/pull/1118